src/cpu-kernels/awkward_NumpyArray_fill_fromcomplex.cpp

```diff
@@ -0,0 +1,193 @@
+// BSD 3-Clause License; see https://github.com/scikit-hep/awkward-1.0/blob/main/LICENSE
+
+#define FILENAME(line) FILENAME_FOR_EXCEPTIONS_C("src/cpu-kernels/awkward_NumpyArray_fill_fromcomplex.cpp", line)
+
+#include "awkward/kernels.h"
+
+template <typename FROM, typename TO>
+ERROR
+awkward_NumpyArray_fill_fromcomplex(TO* toptr,
+                                    int64_t tooffset,
+                                    const FROM* fromptr,
+                                    int64_t length) {
+  for (int64_t i = 0; i < length; i++) {
+    toptr[tooffset + i] = (TO)fromptr[i*2];
```

Comment:
as implemented by @sjperkins in [here](https://github.com/scikit-hep/awkward-1.0/pull/421/files#diff-d42f70fefa3b239e75f4d450c832faed1ec03069fe9b55b8ee8b2fca898d9527R1197) - only real part is kept when converting complex to any other type.